truffle/src/main/java/org/jruby/truffle/nodes/core/TrufflePrimitiveNodes.java

```diff
@@ -456,4 +456,24 @@ public RubyNilClass simpleShell() {
 
     }
 
+    @CoreMethod(names = "debug_print", onSingleton = true, required = 1)
+    public abstract static class DebugPrintNode extends CoreMethodNode {
+
+        public DebugPrintNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public DebugPrintNode(DebugPrintNode prev) {
+            super(prev);
+        }
+
+        @CompilerDirectives.TruffleBoundary
+        @Specialization
+        public RubyNilClass debugPrint(RubyString string) {
```

Comment:
I was wondering why you would need this and then I remembered you were rewriting IO ;-)